aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -93,146 +165,126 @@ module AptosFramework::Token {
+        mint_token_events: EventHandle<MintTokenEvent>
     }
 
     /// Represent the collection metadata
     struct Collection has store {
         // Describes the collection
-        description: ASCII::String,
+        description: String,
         // Unique name within this creators account for this collection
-        name: ASCII::String,
+        name: String,
         // URL for additional information /media
-        uri: ASCII::String,
-        // Total number of distinct Tokens tracked by the collection
+        uri: String,
+        // Total number of distinct TokenData tracked by the collection
         count: u64,
-        // Optional maximum number of tokens allowed within this collections
-        maximum: Option<u64>,
-    }
-
-    /// The royalty of a token
-    struct Royalty has copy, drop, store {
-        // The basis point and denominator is ROYAL_BASIS_POINT_DENOMINATOR (1,000,000)
-        royalty_points_per_million: u64,
-        // if the token is jointly owned by multiple creators,
-        // the group of creators should create a shared account.
-        // the creator_account will be the shared account address.
-        creator_account: address,
+        // maximum number of TokenData allowed within this collections
+        maximum: u64,
+        // control which collection field is mutable
+        mutability_config: CollectionMutabilityConfig
     }
 
-    /// The data associated with the Tokens
-    struct TokenData has copy, drop, store {
-        // Unique name within this creators account for this Token's collection
-        collection: ASCII::String,
-        // Describes this Token
-        description: ASCII::String,
-        // The name of this Token
-        name: ASCII::String,
-        // Optional maximum number of this type of Token.
-        maximum: Option<u64>,
-        // Total number of this type of Token
-        supply: Option<u64>,
-        // URL for additional information / media
-        uri: ASCII::String,
-        // the royalty of the token
-        royalty: Royalty,
+    /// create collection event with creator address and collection name
+    struct CreateCollectionEvent has drop, store {
+        creator: address,
+        collection_name: String,
+        uri: String,
+        description: String,
+        maximum: u64
     }
 
     /// Capability required to mint tokens.
     struct MintCapability has store {
-        token_id: TokenId,
+        token_id: TokenId
     }
 
     /// Capability required to burn tokens.
     struct BurnCapability has store {
-        token_id: TokenId,
+        token_id: TokenId
     }
 
     //
     // Creator Script functions
     //
 
-    public(script) fun create_limited_collection_script(
+    public(script) fun create_collection_script(
         creator: &signer,
         name: vector<u8>,
         description: vector<u8>,
         uri: vector<u8>,
         maximum: u64,
+        mutate_setting: vector<bool>
```

Comment:
mutable_settings

and again, maybe we keep the existing interfaces so you don't have to do all downstream dependencies

Reply:
as mentioned above.